vuu-java/src/test/java/org/finos/vuu/net/rpc/RpcMethodHandlerTest.java

```diff
@@ -0,0 +1,55 @@
+package org.finos.vuu.net.rpc;
+
+import org.assertj.core.api.Assertions;
+import org.finos.vuu.net.LoginRequest;
+import org.finos.vuu.net.RpcCall;
+import org.finos.vuu.net.RpcResponse;
+import org.finos.vuu.net.ViewServerMessage;
+import org.finos.vuu.util.ScalaCollectionConverter;
+import org.junit.Test;
+import scala.Option;
+import test.helper.ViewPortTestUtils;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static test.helper.ViewPortTestUtils.createRandomViewServerMessage;
+
+public class RpcMethodHandlerTest {
+
+    @Test
+    public void should_wrap_and_call_java_function_in_method_handler() {
+        final TestRpcService rpcService = new TestRpcService();
+
+        final RpcMethodHandler handler = new RpcFunctionMethodHandler(rpcService::rpcFunction);
+
+        final RpcMethodCallResult result = handler.call(new RpcParams(new Object[]{}, ScalaCollectionConverter.toScala(Collections.emptyMap()), null));
+
+        assertThat(result)
+                .isNotNull()
+                .isExactlyInstanceOf(RpcMethodSuccess.class)
+                .extracting("result").isEqualTo("It Works");
+    }
+
+    @Test
+    public void should_register_java_function_as_rpc_in_default_handler() {
+        final TestRpcService rpcService = new TestRpcService();
+        final RpcMethodHandler handler = new RpcFunctionMethodHandler(rpcService::rpcFunction);
+
+        final DefaultRpcHandler defaultRpcHandler = new DefaultRpcHandler();
+        defaultRpcHandler.registerRpcMethodHandler("helloWorld", handler);
+
+        RpcCall call = new RpcCall("service", "helloWorld", new Object[]{}, ScalaCollectionConverter.toScala(Collections.emptyMap()));
+        Option<ViewServerMessage> response = defaultRpcHandler.processRpcCall(createRandomViewServerMessage(new LoginRequest("token", "user")), call, ViewPortTestUtils.requestContext());
+
+        Assertions.assertThat(response.get().body())
+                .isExactlyInstanceOf(RpcResponse.class)
+                .isEqualTo(new RpcResponse("helloWorld", new RpcMethodSuccess("It Works"), null));
```

Comment:
I think we should avoid exposing RpcMethodSuccess object on the RcpResponse.

Reply:
In terms of current use case
- login request currently doesnt go through rpc handler, its special rcp message that get handled at central place. - not an issue in terms of unit test, just fyi
- The only current use case of RcpCall -> RcpResponse is the typeahead. Which returns list of values that match the search criteria as string. I guess idea is that each rcp call can return a different result type? So this change essentially will change the response returned from typeahead.
- We plan to change the type ahead to use ViewPortRpcCall -> ViewPortRpcResponse which returns a ViewPortAction. We dont currently have an action that lets you retunr values like we need for type ahead. Not relevant to this change but just fyi as well

Reply:
OK, not exposing MethodSuccess to the response, it will be consistent with the existing API,

When it comes to the further comments, login is just an example message to not create a new one.